### PR TITLE
Sorting server settings in map editor

### PR DESCRIPTION
**Pull request by archimede67**
The server settings are sorted by commands name in alphabetical order

### CODE REVIEW

**Comment by def-** [comment]
Not sure if we want them sorted. I can imagine a mapper grouping the commands and having some comments in there to explain what each group does.

**Comment by Learath2** [comment]
Agreed, we should leave the sorting up to the mapper

**Comment by archimede67** [comment]
So maybe i can add a button to enable/disable sorting?

**Comment by Pulsartw** [comment]
@def- @Learath2 The way it works now is just awful. Sorting would help a lot and definitely won't do any harm to anyone, like seriously where's the problem? It's just a simple solution for grouping/editing/inserting.
@archimede67 Please no more buttons..

**Comment by Learath2** [comment]
Awful? Am I missing something? Sorting helps with what exactly? This is the opposite of a solution for "grouping" as it gives you no way to "group", it doesn't have any advantages "editing" and maybe makes mindlessly inserting a little simpler.

The better solution to this would be to replace the whole "Server settings" part with a crude text editor and let the mapper handle it all.

**Comment by Pulsartw** [comment]
Yes awful. Imagine you're coding and if you want to edit something in the middle of your code, you are forced to remove it and write everything from that point till the end again. Isn't it awful?

Well look, it keeps commands in a certain order, so it groups them in a way, sv_ will be with sv_, tunes will be with tunes, switches with switches etc. I didn't really mean literal editing here, but because the value is inserted always in the same spot, you can kind of 'edit' it by removing and adding it back.

Basically the problem here is the mess when you use many settings and start changing values, sorting helps to keep it clean.
And ye ofc, obviously I know there are better solutions, but I don't see the reason to waste time on something so trivial, so I proposed something simple to implement that would actually help.

**Comment by Learath2** [comment]
Well, currently Kobra 4 e.g. has 2 groups and 2 comments that describe them in it's settings, with this patch that'd be impossible. I'm quite against merging this.

However the experience is indeed pretty dismal, so I would definitely accept a patch that changes this entire UI with a textbox.

**Comment by Pulsartw** [comment]
Aw damn, ok I see where you're coming from now, I got confused because I wanted it to sort values every time Add button is pressed. I guess I either forgot to tell @archimede67 about it or he missed that. Would this be ok?

**Comment by Pulsartw** [comment]
Unless it's a problem that mappers won't be able to write comments anymore.

**Comment by heinrich5991** [comment]
If you want to easily edit the config right now, use the `config_retrieve` and `config_store` tools. Just drop your map onto `config_retrieve`, edit the config with your favorite editor and then drop your map onto `config_store`.

**Comment by 12pm** [comment]
don’t merge please. half assed solution to a rare issue.

**Comment by Pulsartw** [comment]
@heinrich5991 thanks, good to know
